src/app/workflow/dag/wdl-viewer/wdl-viewer.service.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ *    Copyright 2017 OICR
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+import { Injectable } from '@angular/core';
+
+import * as JSZip from 'jszip';
+import * as pipeline from 'pipeline-builder';
+import { Observable, from, forkJoin } from 'rxjs';
+import { switchMap } from 'rxjs/internal/operators';
+import { GA4GHFilesQuery } from '../../../shared/ga4gh-files/ga4gh-files.query';
+import { ExtendedWorkflow } from '../../../shared/models/ExtendedWorkflow';
+import { ToolDescriptor, ToolFile, WorkflowsService, WorkflowVersion } from '../../../shared/swagger';
+
+
+/**
+ * Service for creating WDL workflow visualizations with EPAM Pipeline Builder library
+ */
+@Injectable()
+export class WdlViewerService {
+  private zip: JSZip = new JSZip();
+  constructor(private gA4GHFilesQuery: GA4GHFilesQuery, protected workflowsService: WorkflowsService) {
+  }
+
+  getFiles(descriptorType: ToolDescriptor.TypeEnum): Observable<Array<ToolFile>> {
+    return this.gA4GHFilesQuery.getToolFiles(descriptorType, [ToolFile.FileTypeEnum.PRIMARYDESCRIPTOR,
```

Comment:
Are all the files guaranteed to be returned here? Not sure, just asking.

Reply:
Can you please verify and respond to my previous question?

Reply:
No, the function does not provide a guarantee. It returns an empty Observable array if there are no files in the store.

Currently, the tool files are placed into the store on initialization of the workflow component. I'm using this function under the assumption that the tool files have been/will be stored beforehand. I also provide a null & empty check when subscribing to this observable as an edge case